applications/FluidDynamicsApplication/python_scripts/check_and_prepare_model_process_fluid.py

```diff
@@ -63,6 +63,10 @@ def Execute(self):
         fluid_computational_model_part.AddConditions(list(list_of_ids))
 
         #verify the orientation of the skin (only implemented for tris and tets)
+        if self.main_model_part.NumberOfElements() == 0:
```

Comment:
Also what if there are no elements but there are conditions?

Reply:
Then the check will fail here:

https://github.com/KratosMultiphysics/Kratos/blob/261342507dc747f540bd89d12a8b425a9e715ea5/kratos/processes/tetrahedral_mesh_orientation_check.cpp#L194-L196

In the future we will need to implement a `quadrilateral_mesh_orientation_check` but for now quadrilateral fluid elements' infrastrucure is still very much in development.

Reply:
Since this is a detail of the `TetrahedralMeshOrientationCheck`, I would slightly modify the check:
~~~py
if self.main_model_part.NumberOfElements() > 0:
        geometry = self.main_model_part.Elements.__iter__().__next__().GetGeometry()
        is_simplex = geometry.LocalSpaceDimension() + 1 == geometry.PointsNumber()
        if not is_simplex:
            msg = "Geoemetry is not simplex. Orientation check is only available"
            msg += "for simplex geometries and hence it will be skipped"
            KratosMultiphysics.Logger.PrintWarning(type(self).__name__, msg)
            return
~~~

this way the overall flow is not changed

Reply:
Well this is going to modify the behaviour of the program beyond what this PR was trying to fix (i.e. the error introduced in #9426).

I don't know if the error thrown by `TetrahedralMeshOrientationCheck` was the expected behaviour or not. If the Fluid application required all conditions to be in the same rank as their parent element (which I think is the case), throwing the error would be good.

@rubenzorrilla thoughts?

Reply:
Oh ok, agreed then.

Reply:
but you didn't push changes?

My concern is that now the `TetrahedralMeshOrientationCheck` is not called any more if there are no elements, hence my suggestion

Reply:
Yeah didn't have to do it yesterday. They're up now.